packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -1367,7 +1368,19 @@ class _NestedScrollPosition extends ScrollPosition implements ScrollActivityDele
 
   @override
   void restoreScrollOffset() {
-    if (coordinator.canScrollBody)
+    // If the children of a [NestedScrollView] have [PageStorageKey], the
```

Comment:
I am not sure I understand this condition. For other scrollables its also not required that the ScrollView itself has the PageStorageKey - it can be somewhere else in the hierarchy.

Reply:
Simply require that the ScrollView has PageStorageKey, then it can record and restore its own scroll pixel instead of the NestedScrollView intervention.

Reply:
Yes, we don't have the means to do it.